restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -154,6 +159,101 @@ public ReactiveDocumentService(
+   * for each document will be randomly generated.
+   *
+   * @param db {@link DocumentDB} to write in
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param payload Documents represented as JSON array
+   * @param idPath Optional path to the id of the document in each doc.
+   * @param context Execution content
+   * @return Document response wrapper containing the generated ID.
+   */
+  public Single<MultiDocsResponse> writeDocuments(
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String payload,
+      String idPath,
+      ExecutionContext context) {
+
+    return Single.defer(
+            () -> {
+              // note that if idPath is present, this means we must use the update method
+              // because it could overwrite the existing doc
+              boolean useUpdate = null != idPath;
+
+              // authentication for writing before anything
+              // we need the DELETE scope if we are updating
+              authorizeWrite(db, namespace, collection, Scope.MODIFY);
+              if (useUpdate) {
+                authorizeWrite(db, namespace, collection, Scope.DELETE);
+              }
+
+              // read the root
+              JsonNode root = readPayload(payload);
+
+              // if not array, fail immediately
+              if (!root.isArray()) {
+                throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_WRITE_BATCH_NOT_ARRAY);
+              }
+
+              // keep order with LinkedHashMap
+              LinkedHashMap<String, List<JsonShreddedRow>> documentRowsMap = new LinkedHashMap<>();
+              Optional<JsonPointer> idPointer = DocsApiUtils.pathToJsonPointer(idPath);
+              for (JsonNode documentNode : root) {
+                // validate that the document fits the schema
+                checkSchemaOnWrite(db, namespace, collection, documentNode);
+
+                // get document id
+                String documentId = documentIdResolver().apply(idPointer, documentNode);
+
+                // shred rows
+                List<JsonShreddedRow> rows =
+                    jsonDocumentShredder.shred(documentNode, Collections.emptyList());
+
+                // add to map and make sure we did not have already the same ID
+                if (documentRowsMap.put(documentId, rows) != null) {
+                  // TODO maybe more friendly message
+                  throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_WRITE_BATCH_DUPLICATE_ID);
+                }
+              }
+
+              // generate the Single for each document write
+              List<Single<ResultSet>> writeAll =
+                  documentRowsMap.entrySet().stream()
+                      .map(
+                          entry -> {
+                            if (useUpdate) {
+                              return writeService.updateDocument(
```

Comment:
So the difference between `updateDocument` and `writeDocument` is that updating does a deletion of certain paths before write?

Reply:
yep, and write does not.. We can use write when we have auto generated IDs..